apps/nextjs/src/components/ContextProviders/ChatProvider.tsx

```diff
@@ -65,6 +67,14 @@ export type ChatContextProps = {
   queuedUserAction: string | null;
   queueUserAction: (action: string) => void;
   executeQueuedAction: () => Promise<void>;
+  streamingSection: LessonPlanKeys | undefined;
+  streamingSections: LessonPlanKeys[] | undefined;
+  streamingSectionCompleted: LessonPlanKeys | undefined;
+  sectionRefs: Record<
```

Comment:
The provider now manages the refs to the sections that are displayed. We use this to manage the scroll behaviour.